Rubberduck.Inspections/Concrete/UnreachableCaseInspection.cs

```diff
@@ -0,0 +1,1309 @@
+                    RangeClausesAsText = new List<string>(),
+                    EnumDiscretes = new List<UnreachableCaseInspectionValue>()
+                };
+                CanBeInspected = TryGetChildContext(SelectStmtContext, out SelectExpressionContext);
+            }
+        }
+
+        internal struct CaseClauseDataObject
+        {
+            public ParserRuleContext CaseContext;
+            public List<RangeClauseDataObject> RangeClauseDOs;
+            public ClauseEvaluationResult ResultType;
+
+            public CaseClauseDataObject(ParserRuleContext caseClause)
+            {
+                CaseContext = caseClause;
+                RangeClauseDOs = new List<RangeClauseDataObject>();
+                ResultType = ClauseEvaluationResult.NoResult;
+            }
+        }
+
+        internal struct RangeClauseDataObject
+        {
+            public RangeClauseContext Context;
+            public bool UsesIsClause;
+            public bool IsValueRange;
+            public bool IsConstant;
+            public bool CompareByTextOnly;
+            public string IdReferenceName;
+            public string AsText;
+            public string TypeNameTarget;
+            public string CompareSymbol;
+            public UnreachableCaseInspectionValue SingleValue;
+            public UnreachableCaseInspectionValue MinValue;
+            public UnreachableCaseInspectionValue MaxValue;
+            public ClauseEvaluationResult ResultType;
+            public bool CanBeInspected;
+
+            public RangeClauseDataObject(RangeClauseContext ctxt, string targetTypeName)
+            {
+                Context = ctxt;
+                UsesIsClause = false;
+                IsValueRange = false;
+                IsConstant = false;
+                CanBeInspected = true;
+                CompareByTextOnly = false;
+                IdReferenceName = string.Empty;
+                AsText = ctxt.GetText();
+                TypeNameTarget = targetTypeName;
+                CompareSymbol = CompareTokens.EQ;
+                SingleValue = null;
+                MinValue = null;
+                MaxValue = null;
+                ResultType = ClauseEvaluationResult.NoResult;
+            }
+        }
+
+        public UnreachableCaseInspection(RubberduckParserState state)
+            : base(state, CodeInspectionSeverity.Suggestion) { }
+
+        public override IInspectionListener Listener { get; } =
+            new UnreachableCaseInspectionListener();
+
+        public override CodeInspectionType InspectionType => CodeInspectionType.CodeQualityIssues;
+
+        private static Dictionary<string, string> CompareInversions = new Dictionary<string, string>()
+        {
+            { CompareTokens.EQ, CompareTokens.EQ },
+            { CompareTokens.NEQ, CompareTokens.NEQ },
+            { CompareTokens.LT, CompareTokens.GT },
+            { CompareTokens.LTE, CompareTokens.GTE },
```

Comment:
Truly inverse comparisons would be `{ LT, GTE }, { LTE, GT }`

Reply:
See comment above - the goal is algebraic inversion rather than logical inversion.

Reply:
Using C# 6, I'm pretty sure you could format these as `[CompareTokens.EQ] = CompareTokens.EQ`, etc.

Reply:
@Hosch250 yes!  Updated this and other Dictionaries accordingly